src/test/java/org/sagebionetworks/workers/util/semaphore/WriteReadSemaphoreImplTest.java

```diff
@@ -78,73 +80,167 @@ public void testGetWriteLockProviderWithNullRequest() {
 	}
 
 	@Test
-	public void testGetReadLockProvider() throws Exception {
+	public void testGetReadLock() throws IOException{
+		
+		doReturn(mockReadLock).when(semaphore).createReadLock(any());
 		
-		when(mockCountingSemaphore.attemptToAcquireLock(any(), anyLong(), anyInt(), any())).thenReturn(
-				Optional.of("tokenOne"));
-		when(mockCountingSemaphore.getFirstUnexpiredLockContext(any())).thenReturn(Optional.empty());
 		when(mockCallback.getLockTimeoutSeconds()).thenReturn(maxTimeout);
-
+		readRequest = new ReadLockRequest(mockCallback, context, keys);
+	
 		// call under test
-		try (ReadLock provider = semaphore
```

Comment:
I was thinking that the try with resource interaction would be tested in this class, e.g. verify that mockLock.attemptToAquireLock and mockLock.close are invoked. While the logic of the actual implementation of attemptToAquireLock and close is tested in the Read/WriteLockImplTest.